src/common/ui/ProgressWindow.jsx

```diff
@@ -202,6 +208,14 @@ Zotero.UI.ProgressWindow = class ProgressWindow extends React.PureComponent {
 		// Alert that the item is being saved or has already been saved
 		let alert = this.done ? Zotero.getString("progressWindow_alreadySaved") : `${text} ${target.name}`;
 		document.getElementById("messageAlert").textContent = alert
+		// Tags array contains objects {tag : ""} that may contain duplicate values due to different types.
+		// Unwrap the tag objects and deduplicate tags values.
+		// this.existingTags will be an object of the form {libraryID: [tag1, tag2, ...]}.
+		let dedupedTags = {};
+		Object.entries(tags || []).forEach(([libraryID, tagArr]) => {
+			dedupedTags[libraryID] = [...new Set(tagArr.map(item => item.tag))];
+		});
+		this.existingTags = dedupedTags;
```

Comment:
Seems like this would be more appropriate to do in `updateFromClient()` which does some preprocessing of data for the Progress Window.

Reply:
That makes a lot of sense, yeah. I completely re-did and re-pushed the last commit with the new `<TagsInput>` component that has our own autocomplete popup. I moved this bit of pre-processing to `updateFromClient` from here.